src/pivot-table/hooks/use-items-rendered-handler/utils/fetch-pages.ts

```diff
@@ -0,0 +1,64 @@
+import { debouncer, throttler } from "qlik-chart-modules";
+import type { DataModel, LayoutService, MeasureData, PageInfo, ViewService } from "../../../../types/types";
+import { ScrollDirection } from "../../use-scroll-direction";
+import getColumnPages from "./get-column-pages";
+import getRowPages from "./get-row-pages";
+
+type FetchPages = (
+  dataModel: DataModel,
+  layoutService: LayoutService,
+  measureData: MeasureData,
+  pageInfo: PageInfo,
+  viewService: ViewService,
+  verticalScrollDirection: React.MutableRefObject<ScrollDirection>,
+  horizontalScrollDirection: React.MutableRefObject<ScrollDirection>,
+) => Promise<void>;
+
+export const BUFFER = 10;
+
+export const fetchPages = async (
+  dataModel: DataModel,
+  layoutService: LayoutService,
+  measureData: MeasureData,
+  pageInfo: PageInfo,
+  viewService: ViewService,
+  verticalScrollDirection: React.MutableRefObject<ScrollDirection>,
+  horizontalScrollDirection: React.MutableRefObject<ScrollDirection>,
+) => {
+  const { gridColumnStartIndex, gridRowStartIndex, gridWidth, gridHeight } = viewService;
+  let rowPages: EngineAPI.INxPage[] = [];
+  let columnsPages: EngineAPI.INxPage[] = [];
+
+  if (verticalScrollDirection.current !== ScrollDirection.None) {
+    const backBuffer = verticalScrollDirection.current === ScrollDirection.Backward ? BUFFER : 0;
```

Comment:
```suggestion
    const getBackBuffer = (dir: <DIR_TYPE>) => dir === ScrollDirection.Backward ? BUFFER : 0;
```
and use it in both if checks